### PR TITLE
Implement SDL-0243 Manager Update for Display Capability

### LINKED ISSUE #1154: [SDL 0243] Manager Update for DisplayCapability
Proposal: [Manager Update for DisplayCapability](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0243-manager-update-display-capability.md)

> This proposal adds display capabilities redesign described in [SDL 0216 Widget Support](0216-widget-support.md) (see issue [#664](https://github.com/smartdevicelink/sdl_evolution/issues/664)) to SystemCapabilityManager.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/797

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revisions:
- In addition to`- (nullable SDLWindowCapability *)windowWithID:(NSNumber<SDLInt> *)windowID; `, include `- (nullable SDLWindowCapability *)mainDisplayMainWindowCapabilities;` with property ` defaultMainWindowCapability`
- Change “In order to avoid clashes with the mobile API the enum should not be extended any further without adding the enum values to the mobile API.” to “In order to avoid clashes with the mobile API the enum should only be extended with extreme caution when absolutely necessary or by adding the enum values to the RPC Spec.”

> The proposal .md file was updated to reflect these revisions on 8/22/19.

### PR DESCRIPTION
**Pull request by kshala-ford**
Fixes #1154 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Extend unit tests to confirm display capability/capabilities conversion is as expected.

### Summary
This PR changes SystemCapabilityManager according to SDL-0243 to make new capabilities available and also convert DisplayCapability and DisplayCapabilities objects.

### Changelog
Existing SystemCapabilityTypes are marked as deprecated hence will appear as deprecated if apps are using them.

##### Enhancements
- Adds new capability to the manager  
- Adds conversion of display capability objects for developer's convenience

### Tasks Remaining:
- [x] [Add DISPLAYS to manager]
- [x] [Add Conversion code]
- [x] [Do the unit tests]

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kshala-ford** [comment]
@theresalech The PR is ready for review.

File: base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java [hunk 1]
```diff
@@ -61,18 +74,151 @@
...
+	private void updateCachedDisplayCapabilityList(List<DisplayCapability> newCapabilities) {
+		List<DisplayCapability> oldCapabilities = (List<DisplayCapability>) getCapability(SystemCapabilityType.DISPLAYS);
+
+		if (oldCapabilities == null) {
+			setCapability(SystemCapabilityType.DISPLAYS, newCapabilities);
+			updateDeprecatedDisplayCapabilities();
+			return;
+		}
+
+		DisplayCapability oldDefaultDisplayCapabilities = oldCapabilities.get(0);
```
[the remark this answers is not in the document]
Comment (kshala-ford) [reply base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java]: made sure in the condition above.

File: base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java [hunk 2]
```diff
@@ -61,18 +74,151 @@
...
+		ArrayList<WindowCapability> copyWindowCapabilities = new ArrayList<>(oldDefaultDisplayCapabilities.getWindowCapabilities());
+
+		DisplayCapability newDefaultDisplayCapabilities = newCapabilities.get(0);
```
[the remark this answers is not in the document]
Comment (kshala-ford) [reply base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java]: Made sure in the condition above

**Comment by kshala-ford** [comment]
@BrettyWhite we attempted the change in the library to use the new format but had issues to get it running stable. The risk of creating bugs is extremely high for actually just little to no benefit as the converted capabilities are compatible.

**Comment by BrettyWhite** [comment]
Closed as livio taking over implementation. Will open new PR

**Pull request closed without merging**